lib/core/macros.nim

```diff
@@ -1433,6 +1433,13 @@ else:
     else:
       result = false
 
+proc expectIdent*(n: NimNode, name: string) {.compileTime, since: (1,1).} =
+  ## Check that ``eqIdent(n,name)`` holds true. If this is not the
+  ## case, compilation aborts with an error message. This is useful
+  ## for writing macros that check the AST that is passed to them.
+  if not eqIdent(n, name):
+    error("macro expects an identifier `" & name & "` here", n)
```

Comment:
```suggestion
    error("Expected identifier to be `" & name & "` here", n)
```
And maybe it should show the identifier of n too?

Reply:
No, printing the name of `n` is not required, since the error message will point directly to where `n` is in the source code.